### PR TITLE
Finish remember me

### PR DESCRIPTION
**Pull request by manhnhhedspi**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ def create
     user = User.find_by email: params[:session][:email].downcase
     if user && user.authenticate(params[:session][:password])
       log_in user
+      params[:session][:remember_me] == "1" ? remember(user) : forget(user)
```
Comment (phuongbk) [inline app/controllers/sessions_controller.rb]: remove to Setting

File: app/models/user.rb [hunk 2]
```diff
@@ -23,4 +24,31 @@ class User < ActiveRecord::Base
     length: {minimum: Settings.user.minimum_characters}
   validates :name, presence: true,
     length: {maximum: Settings.user.maximum_characters}
+
+  def self.digest string
+    if ActiveModel::SecurePassword.min_cost
```
Comment (dieunb) [inline app/models/user.rb]:
```
+    cost = if ActiveModel::SecurePassword.min_cost
+      BCrypt::Engine::MIN_COST
+    else
+      BCrypt::Engine.cost
+    end
```

File: app/models/user.rb [hunk 3]
```diff
@@ -23,4 +24,31 @@ class User < ActiveRecord::Base
...
+    BCrypt::Password.create string, cost: cost
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attributes :remember_digest => User.digest(remember_token)
```
Comment (dieunb) [inline app/models/user.rb]: convention

File: app/models/user.rb [hunk 4]
```diff
@@ -23,4 +24,31 @@ class User < ActiveRecord::Base
...
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attributes :remember_digest => nil
```
Comment (dieunb) [inline app/models/user.rb]: convention

**Pull request merged**